qiskit/compiler/transpile.py

```diff
@@ -308,13 +353,21 @@ def _transpile_circuit(circuit_config_tuple: Tuple[QuantumCircuit, Dict]) -> Qua
     if ms_basis_swap is not None:
         pass_manager.append(MSBasisDecomposer(ms_basis_swap))
 
+    if pass_manager_config.scheduling_method is not None:
+        if 'delay' not in pass_manager_config.basis_gates:
+            pass_manager_config.basis_gates.append('delay')
+
+        from qiskit.transpiler.passes.scheduling import RemoveOpsOnIdleQubits
+        pass_manager.append(RemoveOpsOnIdleQubits())
```

Comment:
why is this needed?

Reply:
To make [this test](https://github.com/Qiskit/qiskit-terra/blob/f1a635a5d149df71db27d8637a73dc47990c97af/test/python/compiler/test_sequencer.py#L46) successful.